apps/site/lib/site_web/controllers/fare_transformation_controller.ex

```diff
@@ -37,7 +37,9 @@ defmodule SiteWeb.FareTransformationController do
         # Return the 10 closest locations, sorted by distance
         nearby_proposed_locations
         |> Util.Distance.sort(position)
-        |> Enum.slice(0, 10)
+        # There are 1178 PSLs. Since we are rendering them as hidden and showing/hiding them based on the filter,
+        # we should really narrow down how many we are passing 
+        |> Enum.slice(0, 50)
```

Comment:
Why are you selecting 50 if we're only showing 10?

Reply:
Let's say there's only 1 "Charlie Retailer" in the top 10.  Then when the user filters by "Charlie Retailer", only 1 PSL will appear (even though there are plenty other Charlie Retailers just a little further away).  

I use a filter in the JS file to narrow down the 50 to the top 12

Reply:
It's a little arbitrary and not especially sophisticated, but absolutely helpful and in my opinion adequate (at least for now)! 😄

Reply:
Haha oh yeah, not at all sophisticated.  But when JS is disabled and can't filter down the list, it does prevent us from rendering 1100+ cards  😵

Reply:
In that case, I think we should filter in the backend and not in JS...